shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/screen/presentation/StudyPlanScreenFeature.kt

```diff
@@ -0,0 +1,57 @@
+package org.hyperskill.app.study_plan.screen.presentation
+
+import org.hyperskill.app.analytic.domain.model.AnalyticEvent
+import org.hyperskill.app.config.BuildKonfig
+import org.hyperskill.app.core.domain.BuildVariant
+import org.hyperskill.app.gamification_toolbar.presentation.GamificationToolbarFeature
+import org.hyperskill.app.study_plan.widget.presentation.StudyPlanWidgetFeature
+
+object StudyPlanScreenFeature {
```

Comment:
1. Why do we have several study plan features: `StudyPlanScreenFeature` and `StudyPlanWidgetFeature`?
2. What purpose of the `StudyPlanScreenFeature`?
3. What purpose of the `StudyPlanWidgetFeature`?

Reply:
@ivan-magda 

`StudyPlanWidgetFeature` is a widget that contains the logic of `StudyPlan`: how to load `StudyPlan`, how to handle user interaction with it, etc.

`StudyPlanScreenFeature` is a screen feature, which is just a host for other features.
It sends its own view analytics event and could delegate screen events such as `SwipeRefresh` to child features.
It is just a box for other features, e.g. SudyPlanWidget, GamificationToolbar, or ProblemsLimit.

The reason, why I separated `StudyPlanWidgetFeature`  and `StudyPlanScreenFeature` is to not mix complex `StudyPlanWidgetFeature.State` with other features states.